### ISSUE TITLE
`cargo test` does not run due to tls conflict

### ISSUE THREAD

**Issue by iajoiner** [delta-io/delta-rs#985]
# Environment

**Delta-rs version**:
Commit 4efc67c718121dd3b5512a772e4036b640cc62ff
**Binding**: 

**Environment**:
- **Cloud provider**:
- **OS**: Ubuntu 22.04
- **Other**:

***
# Bug

**What happened**:
```
error[E0252]: the name `tls` is defined multiple times
  --> /home/iajoiner/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.48.0/src/lib.rs:25:5
   |
23 | use hyper_rustls as tls;
   |     ------------------- previous import of the module `tls` here
24 | #[cfg(feature = "native-tls")]
25 | use hyper_tls as tls;
   |     ^^^^^^^^^^^^^^^^ `tls` reimported here
   |
   = note: `tls` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
   |
25 | use hyper_tls as other_tls;
   |     ~~~~~~~~~~~~~~~~~~~~~~

error[E0599]: no function or associated item named `new` found for struct `hyper_rustls::HttpsConnector` in the current scope
   --> /home/iajoiner/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.48.0/src/request.rs:224:41
    |
224 |         let connector = HttpsConnector::new();
    |                                         ^^^ function or associated item not found in `hyper_rustls::HttpsConnector<_>`

error[E0599]: no function or associated item named `new` found for struct `hyper_rustls::HttpsConnector` in the current scope
   --> /home/iajoiner/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.48.0/src/request.rs:244:41
    |
244 |         let connector = HttpsConnector::new();
    |                                         ^^^ function or associated item not found in `hyper_rustls::HttpsConnector<_>`

```
**What you expected to happen**:

**How to reproduce it**:
Run `cargo test` at the top level

**More details**:
The essence of the problem is that we are using an unofficial AWS SDK crate, rusoto, that has been in maintenance mode since 2020 when there is in fact an official AWS SDK crate, https://github.com/awslabs/aws-sdk-rust now. Since rusoto has two mutually exclusive features without adding a compiler error to detect them being used in combination we have this bug.

**Comment by iajoiner** [comment]
OK this is a duplicate of https://github.com/delta-io/delta-rs/issues/677. We need to fix this and add the top level CI so that any future regression will be detected.
